File: app/src/main/java/com/pens/it/d4b2018/clara_android/MainActivity.java
package com.pens.it.d4b2018.clara_android;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dashboard_layout);
//        setContentView(R.layout.profile_layout);
//
//        setHeaderTitle("Asset");
//
//        setCurrentFragment(new TestFragment());
    }

//    private void setCurrentFragment(Fragment fragment) {
//        getSupportFragmentManager()
//                .beginTransaction()
//                .add(R.id.main_frame_container, fragment)
//                .commit();
//    }
//
//    private void setHeaderTitle(String title) {
//        TextView mainTitleTV = findViewById(R.id.profile_title_textview);
//        mainTitleTV.setText(title);
//    }

}

File: app/src/main/java/com/pens/it/d4b2018/clara_android/mvvm/view/main/MainActivity.kt
package com.pens.it.d4b2018.clara_android.mvvm.view.main

import android.os.Bundle
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatTextView
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.pens.it.d4b2018.clara_android.R
import com.pens.it.d4b2018.clara_android.mvvm.view.utils.visible

class MainActivity : AppCompatActivity() {

    private lateinit var contentFrameLayout: FrameLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initializeView()
        setDashboardFragment()

        val navigation = findViewById<BottomNavigationView>(R.id.bottom_navigation_bar)
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)
    }

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.home_page_item_menu -> { setDashboardFragment() }
            R.id.reservation_page_item_menu -> { setReservationsFragment() }
            R.id.asset_page_item_menu -> { setAssetsFragment() }
            R.id.profile_page_item_menu -> { setProfileFragment() }
        }

        true
    }

    private fun setCurrentFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction().apply {
            replace(R.id.content_frame_layout, fragment)
            commit()
        }
    }
    
    private fun initializeView() {
        setContentView(R.layout.activity_main)
        contentFrameLayout = findViewById(R.id.content_frame_layout)
        contentFrameLayout.visible(true)
    }

    private fun setDashboardFragment() {
        val dashboardFragment = DashboardFragment.newInstance()
        setCurrentFragment(dashboardFragment)
    }

    private fun setAssetsFragment() {
        val assetsFragment = AssetsFragment.newInstance()
        setCurrentFragment(assetsFragment)
    }

    private fun setReservationsFragment() {
        val reservationsFragment = ReservationsFragment.newInstance()
        setCurrentFragment(reservationsFragment)
    }

    private fun setProfileFragment() {
        val profileFragment = ProfileFragment.newInstance()
        setCurrentFragment(profileFragment)
    }
}